### DISCUSSION TITLE
AOC Day 10 Discussion -  Cathode-Ray Tube 🖥

### DISCUSSION THREAD

**Discussion by doleraj** [Virtual-Coffee/virtualcoffee.io#769] [Advent of Code 2022]
Discussion of solutions for [Advent of Code Day 10](https://adventofcode.com/2022/day/10) here.

---
Don't forget about the VC Advent of Code private leaderboard:

Go to [leaderboard/private](https://adventofcode.com/2022/leaderboard/private)
Use join code 363496-e6f42283

**Comment by doleraj** [comment]
What a breath of fresh air after yesterday.

Downside, the aocrunner doesn't work for this sort of problem! But we worked around it.
https://github.com/doleraj/adventofcode/blob/main/aoc2022/src/day10/index.ts

**Reply by danieltott** [reply]
Nice - we thought along the same lines there.

I spent a bit trying to think of a clever way to parse the list initially to make the internal logic more straightforward..

Seeing yours gave me an idea:

```js
const input = `noop
addx 22
addx -17
addx 1
addx 4`

input.split(/\n| /).map(line => isNaN(line) ? 'next' : parseInt(line))

// ["next", "next", 22, "next", -17, "next", 1, "next", 4]
```

**Reply by doleraj** [reply]
Oh, I like it!

**Comment by danieltott** [comment]
Wooooo finally made it

https://github.com/danieltott/aoc2022/blob/main/src/day10/index.ts

As always, at least 50% of my troubles were all reading comprehension. I spent a good ten or twenty minutes thinking the row length was `20` instead of `40` for part 2

<img src="https://media1.giphy.com/media/E2USislQIlsfm/giphy.gif"/>

**Reply by rhdeck** [reply]
[Fancy hulk](https://playgroundai.com/post/clbi9soin07rgs60144v9uwyp) is proud of you, @danieltott .

**Reply by doleraj** [reply]
I feel like this was intentional, with the +20 offset in part 1.

**Comment by rhdeck** [comment]
List comprehensions again for what felt like an economical (at least in line count) [solve](https://github.com/rhdeck/aoc-2022/blob/main/src/10/day_10.ipynb)

**Reply by tkshill** [reply]
Spectacular!

**Comment by tkshill** [comment]
Day 9 took me two days.

Day 10 took me an hour. 😑  Scan and parse and scan and parse.

[solution](https://github.com/tkshill/AdventOfCode2022/blob/main/AOC2022/Day10.fs)

**Reply by davidalpert** [reply]
an hour? I'm impressed. I took about 5 days simply to understand Day 10 part 2

we really should pair on these because you approach them so cleanly

**Reply by davidalpert** [reply]
and I was impressed with how concise _my_ parser is:
- https://github.com/davidalpert/advent-of-code-da/blob/2022/2022/day10_cathode_ray_tube.fs#L12-L27

then I look at yours 🤣

**Comment by davidalpert** [comment]
here's my Day 10, with many thanks to @doleraj for talking me through how to even even understand the Part 2 description (which I had to rewrite so that I could really parse it):
- https://github.com/davidalpert/advent-of-code-da/blob/2022/2022/day10_cathode_ray_tube.fs

F#'s `Seq.unfold` is [kind of magical](https://github.com/davidalpert/advent-of-code-da/blob/2022/2022/day10_cathode_ray_tube.fs#L60-L84) for generating sequences once you wrap your mind around it, but I have to re-wrap my mind around it every time I find a use for it
